src/v/cluster/client_quota_serde.h

```diff
@@ -164,32 +166,118 @@ struct entity_value
     std::optional<uint64_t> controller_mutation_rate;
 };
 
+/// entity_value_diff describes the quotas diff for an entity_key
+struct entity_value_diff
+  : serde::
+      envelope<entity_value_diff, serde::version<0>, serde::compat_version<0>> {
+    enum key : int8_t {
+        producer_byte_rate = 0,
+        consumer_byte_rate,
+        controller_mutation_rate,
+    };
+
+    enum operation : int8_t {
```

Comment:
you can drop `int8_t` -- serde will always store them as 32-bit integers.

Reply:
Done -- https://github.com/redpanda-data/redpanda/pull/20802